assets/css/admin/style.scss

```diff
@@ -93,12 +93,33 @@ body.post-type-download #screen-meta {
 	padding: 0 20px;
 }
 
+#edd-header-branding {
+	vertical-align: middle;
+}
+
 #edd-header img {
-	display: block;
```

Comment:
Removing this line makes the image layout be inline, which visually adds a little extra space below the image. Setting it to block was intentional to have the padding on the top and bottom of the header be even.